includes/Core/Util/Entity_Factory.php

```diff
@@ -60,8 +60,54 @@ public static function from_context() {
 
 		// Otherwise, run frontend-specific `WP_Query` logic.
 		if ( $wp_the_query instanceof WP_Query ) {
-			return self::from_wp_query( $wp_the_query );
+			$entity = self::from_wp_query( $wp_the_query );
+
+			if ( null === $entity ) {
+				return null;
+			}
+
+			if ( ! defined( 'AMP__VERSION' ) ) {
+				return $entity;
+			}
```

Comment:
Let's combine these two conditions into one:

```suggestion
			if ( is_null( $entity ) || ! defined( 'AMP__VERSION' ) ) {
				return $entity;
			}
```